### PR TITLE
Add Apache 2.0 licence to Synapse Team templated projects

### PR DESCRIPTION
**Pull request by reivilibre**

### CODE REVIEW

**Comment by callahad** [comment]
Also add the SPDX value (`Apache License 2.0`) to setup.cfg?

File: {{ cookiecutter.directory_name }}/setup.cfg [hunk 1]
```diff
@@ -6,6 +6,9 @@ version = 0.0.0
 install_requires =
   attrs
 
+classifiers =
+  License :: OSI Approved :: Apache Software License
+
```
Comment (callahad) [inline {{ cookiecutter.directory_name }}/setup.cfg:11]: I think we also want `license = Apache License 2.0` in addition to the Trove classifier, per example at https://setuptools.pypa.io/en/latest/userguide/declarative_config.html

Comment (callahad) [reply {{ cookiecutter.directory_name }}/setup.cfg:11]: Or maybe not per https://www.python.org/dev/peps/pep-0314/#license

Comment (reivilibre) [reply {{ cookiecutter.directory_name }}/setup.cfg:11]: My understanding is that usage of the `license` key is frowned upon for standard licences and the classifiers are preferred: https://setuptools.pypa.io/en/stable/deprecated/distutils/setupscript.html?highlight=license#additional-meta-data.

Comment (callahad) [reply {{ cookiecutter.directory_name }}/setup.cfg:11]: Yeah, https://github.com/raimon49/pip-licenses/issues/60 is also compelling. Leave it as Trove.

File: {{ cookiecutter.directory_name }}/tests/test_example.py [hunk 2]
```diff
@@ -1,3 +1,19 @@
+{% if cookiecutter.variant == "synapse_team" -%}
+# Copyright 2021 The Matrix.org Foundation C.I.C.
```
Comment (DMRobertson) [inline {{ cookiecutter.directory_name }}/tests/test_example.py:2]: Could change 2021 to $CURRENT_YEAR if you're feeling bold

**Review by DMRobertson — APPROVED** [review]
Seems reasonable to me. You and Dan have covered how to mark the licence above.

File: {{ cookiecutter.directory_name }}/tests/test_example.py (hunk 2 above)
Comment (babolivier) [reply {{ cookiecutter.directory_name }}/tests/test_example.py:2]: According to https://github.com/cookiecutter/cookiecutter/issues/902 we could do that with Jinja, which comes with cookiecutter.

Comment (reivilibre) [reply {{ cookiecutter.directory_name }}/tests/test_example.py:2]: I considered that, but not sure it's necessarily valid until you actually add a copyrightable change to the templated code. But also, you're very likely to do so if using the template anyway, so maybe I should?

Comment (babolivier) [reply {{ cookiecutter.directory_name }}/tests/test_example.py:2]: Yeah I think we should

**Pull request merged**